packages/remark-extend/README.md

```diff
@@ -0,0 +1,314 @@
+
+Some examples are:
+
+- `:root` for the top of the markdown file
+- `:scope:last-child` for the end of the markdown file
+- `heading:has([value=Red])` first heading with a text value of Red (e.g. ### Red)
+- `heading[depth=2]` first third level heading (e.g. ## Something)
+
+### Markdown AST
+
+All adjustments to the markdown file happen via the markdown AST (Abstract Syntax Tree).
+
+You can explore it via the [ASTExplorer](https://astexplorer.net/).
+
+```md
+### Red
+
+red is the fire
+```
+
+Resulting AST.
+
+```json
+{
+  "type": "root",
+  "children": [
+    {
+      "type": "heading",
+      "depth": 3,
+      "children": [
+        {
+          "type": "text",
+          "value": "Red"
+        }
+      ]
+    },
+    {
+      "type": "paragraph",
+      "children": [
+        {
+          "type": "text",
+          "value": "red is the fire"
+        }
+      ]
+    }
+  ]
+}
+```
+
+---
+
+### Replacement
+
+Does adjustments from a start point downwards.
+The provided function does get called for every node from the starting point (including the starting point).
+
+#### Replacement Input File
+
+```md
+### Red
+
+red is the fire
+```
+
+#### Replacement Extending File
+
+Goal is to replace all red with green.
+
+````md
+```js ::replaceFrom(':root')
+module.exports.replaceSection = node => {
+  if (node.value) {
+    node.value = node.value.replace(/red/g, 'green').replace(/Red/g, 'Green');
+  }
+  return node;
+};
+```
+````
+
+This function gets called with these nodes in order.
+
+1. root
+2. heading
+3. text
+4. paragraph
+5. text
+
+#### Replacement Result
+
+```md
+### Green
+
+green is the fire
+```
+
+### Replacement Range
+
+Whenever a section or part of the original markdown needs to be adjusted.
+The function does get every node from the starting point (including the starting point) till the end point (excluding the end point).
+
+#### Replacement Range Input File
+
+```md
+### Red <-- starting point (including)
+
+red is the fire
+
+### More Red <-- end point (excluding)
+
+the sun can get red
+```
+
+#### Replacement Range Extending File
+
+````md
+```js ::replaceBetween('heading:has([value=Red])', 'heading:has([value=More Red])')
+module.exports.replaceSection = node => {
+  if (node.value) {
+    node.value = node.value.replace(/red/g, 'green').replace(/Red/g, 'Green');
+  }
+  return node;
+};
+```
+````
+
+#### Replacement Range Result
+
+```md
+### Green <-- starting point (including)
+
+green is the fire
+
+### More Red <-- end point (excluding)
+
+the sun can get red
+```
+
+---
+
+### Add More Markdown Content After
+
+If additional markdown content should be inserted after at a specific location.
+
+#### Add After Input File
+
+```md
+### Red
+
+red is the fire
+```
+
+#### Add After Extending File
+
+````md
+```
+::addMdAfter('heading:has([value=Red])')
+```
+
+the ocean is blue
+````
+
+#### Add After Result
+
```

Comment:
yeah that will not work... it will be like this
- Add something at the end of a "section": `::addMdBefore('heading:has([value=Red]) ~ heading[depth=3]')`
  It works by selecting the headline of your section and add before the next sibling headline with the same depth.

Reply:
Aah okay makes sense